src/kakarot/accounts/library.cairo

```diff
@@ -448,6 +454,22 @@ namespace internal {
         return ();
     }
 
+    // @notice asserts that the new version is greater than the current one
+    // @param new_version The new version of the class.
+    // @param old_version The previous version of the class.
+    func assert_version_upgrade{range_check_ptr}(new_version: felt) {
```

Comment:
why wouldn't we want to downgrade an account?

Reply:
We don't want users to upgrade to a previous version that might be deprecated

Reply:
to me it's more a UX question than a core limitation we want to introduce, let's ask what the other thinks about this, but really doesn't seem obvious to me that we want to enforce this